examples/webgl_geometry_teapot.html

```diff
@@ -0,0 +1,383 @@
+				};
+
+				var h;
+
+				var gui = new dat.GUI();
+
+				// material (attributes)
+
+				h = gui.addFolder( "Material control" );
+
+				h.add( effectController, "shininess", 1.0, 400.0, 1.0 ).name("Shininess");
+				h.add( effectController, "ka", 0.0, 1.0, 0.025 ).name("Ka");
+				h.add( effectController, "kd", 0.0, 1.0, 0.025 ).name("Kd");
+				h.add( effectController, "ks", 0.0, 1.0, 0.025 ).name("Ks");
+				h.add( effectController, "metallic" );
+
+				// material (color)
+
+				h = gui.addFolder( "Material color" );
+
+				h.add( effectController, "hue", 0.0, 1.0, 0.025 ).name("Hue");
+				h.add( effectController, "saturation", 0.0, 1.0, 0.025 ).name("Saturation");
+				h.add( effectController, "lightness", 0.0, 1.0, 0.025 ).name("Lightness");
+
+				// light (point)
+
+				h = gui.addFolder( "Light color" );
+
+				h.add( effectController, "lhue", 0.0, 1.0, 0.025 ).name("Hue");
+				h.add( effectController, "lsaturation", 0.0, 1.0, 0.025 ).name("Saturation");
+				h.add( effectController, "llightness", 0.0, 1.0, 0.025 ).name("Lightness");
+
+				// light (directional)
+
+				h = gui.addFolder( "Light direction" );
+
+				h.add( effectController, "lx", -1.0, 1.0, 0.025 ).name("x");
+				h.add( effectController, "ly", -1.0, 1.0, 0.025 ).name("y");
+				h.add( effectController, "lz", -1.0, 1.0, 0.025 ).name("z");
+
+				h = gui.addFolder( "Tessellation control" );
+				h.add(effectController, "newTess", [2,3,4,5,6,8,10,15,20,30,40,50] ).name("Tessellation Level");
+				h.add(effectController, "lid").name("display lid");
+				h.add(effectController, "body").name("display body");
+				h.add(effectController, "bottom").name("display bottom");
+				h.add(effectController, "fitLid").name("snug lid");
+				h.add(effectController, "nonblinn").name("original scale");
+
+				// shading
+				h = gui.add(effectController, "newShading", ["wireframe","flat","smooth","glossy","textured","reflective"] ).name("Shading");
+			}
+
+
+			//
+
+			function animate() {
+
+				requestAnimationFrame( animate );
+				render();
+				stats.update();
+
+			}
+
+			function render() {
+
+				var delta = clock.getDelta();
+
+				cameraControls.update( delta );
+
+				if ( effectController.newTess !== tess ||
+					effectController.bottom !== bBottom ||
+					effectController.lid !== bLid ||
+					effectController.body !== bBody ||
+					effectController.fitLid !== bFitLid ||
+					effectController.nonblinn !== bNonBlinn ||
+					effectController.newShading !== shading)
+				{
+					tess = effectController.newTess;
+					bBottom = effectController.bottom;
+					bLid = effectController.lid;
+					bBody = effectController.body;
+					bFitLid = effectController.fitLid;
+					bNonBlinn = effectController.nonblinn;
+					shading = effectController.newShading;
+
+					fillScene();
+				}
+
+				phongMaterial.shininess = effectController.shininess;
+				texturedMaterial.shininess = effectController.shininess;
+
+				var diffuseColor = new THREE.Color();
```

Comment:
I'd avoid instantiating a new `THREE.Color` inside the render loop. Create one and reuse it.